### PR TITLE
Fixed onKeyDownQuestionMark for the SingleDatePicker

### PR DESCRIPTION
**Pull request by erin-doyle**
Being able to hit the question mark key in the input of a SingleDatePicker to display the Keyboard Shortcuts panel was not working.  Now it is all connected up and in parity with the DateRangePicker.

I also cleaned up some small naming inconsistencies on the DateRangePicker side and a couple copy/paste mistakes from DateRange to SingleDate, hope that's ok!

### CODE REVIEW

File: src/components/DateRangePickerInput.jsx [hunk 1]
```diff
@@ -39,8 +39,8 @@ const propTypes = forbidExtraProps({
   onStartDateShiftTab: PropTypes.func,
   onEndDateTab: PropTypes.func,
   onClearDates: PropTypes.func,
-  onArrowDown: PropTypes.func,
-  onQuestionMark: PropTypes.func,
```
Comment (ljharb) [inline src/components/DateRangePickerInput.jsx]: this makes it a breaking change, unfortunately

Comment (erin-doyle) [reply src/components/DateRangePickerInput.jsx]: (doh) sorry I should have known that.  I can split that out, yes.

File: src/components/SingleDatePicker.jsx [hunk 2]
```diff
@@ -211,20 +212,22 @@ class SingleDatePicker extends React.Component {
     });
 
     onFocusChange({ focused: false });
-    onClose({ startDate, endDate });
+    onClose({ date });
```
Comment (ljharb) [inline src/components/SingleDatePicker.jsx]: hm, since `onClose` is user code, this is a breaking change as well.

Comment (erin-doyle) [reply src/components/SingleDatePicker.jsx]: Bugger.  Ok I'll move it over!

**Comment by ljharb** [comment]
This should be rebased; and unless @majapw objects, it'll be ready to merge!

**Comment by erin-doyle** [comment]
Awesome thanks happy to help! 👍 🖖

**Pull request merged**